recipe-details.js

```diff
@@ -110,144 +115,238 @@ function setPreparationSteps(recipe) {
-        );
+      if (ingredientsInput.length < 5) {
+        alert("Please provide at least 5 ingredients.");
         return;
       }
 
-      const name = parts[0].trim();
-      const amount = parts[1].trim();
+      const ingredients = [];
+      ingredientsInput.forEach((ingredientLine) => {
+        const parts = ingredientLine.split(":");
 
-      ingredients.push({ NAME: name, AMOUNT: amount });
-    });
+        if (parts.length !== 2) {
+          alert(
+            "Invalid ingredient format. Please use 'Ingredient Name: Amount' format."
+          );
+          return;
+        }
 
-    const newRecipeData = {
-      id: recipes.length + 1,
-      title,
-      imageUrl,
-      description,
-      ingredients,
-      preparationSteps,
-    };
-    console.log(recipes);
-    addNewRecipe(newRecipeData);
-    document.getElementById("title").value = "";
-    document.getElementById("image-url").value = "";
-    document.getElementById("description").value = "";
-    document.getElementById("ingredients").value = "";
-    document.getElementById("preparation-steps").value = "";
-  });
-}
+        const name = parts[0].trim();
+        const amount = parts[1].trim();
 
-// Handle Home link and Recipes Grid links
-function setNavBar() {
-  const setHome = document.getElementById("nav-home");
-  if (setHome) {
-    setHome.addEventListener("click", () => {
-      window.location.href = "index.html";
+        ingredients.push({ NAME: name, AMOUNT: amount });
+      });
+
+      const newRecipeData = {
+        id: recipes.length + 1,
+        title,
+        imageUrl,
+        description,
+        ingredients,
+        preparationSteps,
+      };
+      console.log(recipes);
+      addNewRecipe(newRecipeData);
+
+      document.getElementById("title").value = "";
+      document.getElementById("image-url").value = "";
+      document.getElementById("description").value = "";
+      document.getElementById("ingredients").value = "";
+      document.getElementById("preparation-steps").value = "";
     });
   }
+}
 
-  const gridOfAllRecipes = document.getElementById("nav-recipes");
-  if (gridOfAllRecipes) {
-    gridOfAllRecipes.addEventListener("click", () => {
-      localStorage.removeItem("filteredRecipes");
-      window.location.href = "recipe-grid.html";
-    });
+// Track Time Spent on page;
+function trackTimeSpent() {
+  const timeSpentOnPageSpan = document.getElementById("time-spent-on-page");
+  const startTime = new Date();
+
+  function updateTime() {
+    const currentTime = new Date();
+    const timeDifference = currentTime - startTime;
+    const totalSeconds = Math.floor(timeDifference / 1000);
+    const minutes = Math.floor(totalSeconds / 60);
+    const seconds = totalSeconds % 60;
+
+    timeSpentOnPageSpan.textContent = `${minutes} : ${seconds}`;
   }
+  setInterval(updateTime, 1000);
 }
+// set cooking timer
+function setCookingTimer() {
```

Comment:
setCookingTimer is very big a contains multiple nested functions. Generally avoid nesting functions unless there is a clear reason to as it can make the code slightly harder to understand.